"PUT mapping with _doc on an index that has types" fails

This test fails twice after it was backported into 6.x today. 

We expect the test fails with "the final mapping would have more than 1 type" but the actual reason is "Document mapping type name can't start with '_', found: [_doc]".

```
FAILURE 0.29s | MixedClusterClientYamlTestSuiteIT.test {p0=indices.put_mapping/20_mix_typeless_typeful/PUT mapping with _doc on an index that has types} <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: Failure at [indices.put_mapping/20_mix_typeless_typeful:73]: the error message was expected to match the provided regex but didn't
   > Expected: the final mapping would have more than 1 type
   >      but: was "{root_cause=[{type=invalid_type_name_exception, reason=Document mapping type name can't start with '_', found: [_doc], stack_trace=InvalidTypeNameException[Document mapping type name can't start with '_', found: [_doc]]\n\tat 
```

I can reproduce this after around 10 runs.

```
./gradlew :qa:mixed-cluster:v6.0.1#mixedClusterTestRunner   \
-Dtests.seed=A3A03AA8BC85CA20   \
-Dtests.class=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT   \
-Dtests.method="test {p0=indices.put_mapping/20_mix_typeless_typeful/PUT mapping with _doc on an index that has types}"   \
-Dtests.security.manager=true   \
-Dtests.locale=de-GR   \
-Dtests.timezone=Asia/Baghdad   \
-Dcompiler.java=11   \
-Druntime.java=8
```

Relates #37871

CI: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+internalClusterTest/284/console

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/indices.put_mapping/20_mix_typeless_typeful.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/indices.put_mapping/20_mix_typeless_typeful.yml
@@ -57,8 +57,8 @@
 "PUT mapping with _doc on an index that has types":
 
  - skip:
-      version: "all"
-      reason: 5.x indices can have types that start with an `_` # AwaitsFix: https://github.com/elastic/elasticsearch/issues/38202
+      version: " - 6.1.99"
+      reason: _doc as a type name was only allowed as of 6.2
 
 
  - do:

